src/main/java/startsteps/ECommerceShop/config/SecurityConfig.java

```diff
@@ -47,7 +47,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                         .requestMatchers("/api/v1/admin",
                                 "/products/add",
                                 "products/delete/{id}",
-                                "products/{productId}").hasAuthority("ADMIN")
+                                "products/{productId}"
+                                ,"products/update/{productId}").hasAuthority("ADMIN")
```

Comment:
you already have the `@Preauthorize` annotation on your controller function so no need to have it here

Reply:
ok, done!